scripts/vesting.ts

```diff
@@ -0,0 +1,83 @@
+import BigNumber from "bignumber.js"
+import * as data from './data/rewards.json'
+
+type StaticVestingInfo = {
+    origin: string
+}
+
+type DynamicVestingInfo = {
+    destination: string,
+    schedule: {
+        amtToBeVestedOverTime: string,
+        start: string,
+        period: string,
+        per_period: string,
+        period_count: string 
+    }
+}
+
+type VestingBatch = {
+    statics: StaticVestingInfo
+    vestings: DynamicVestingInfo[]
+}
+
+const origin = '!!! address that pays bsx as they vest; treasury address I guess !!!' 
+// todo;  this is the address that pays out bsx as they vest, to target
+
+const thirtyPercentStart = '!!! end block of BSX LBP !!!'  
+// end block of BSX LBP, when is that ?
+
+const seventyPercentStart = '9071999'  
+// todo fetch; start block of lease ^^^ instead of hard-code
+
+const seventyPercentEnd = '13910400'  
+// todo fetch and calculate; end block of lease ^^^ instead of hard-code
+
+const leaseDuration = Number(seventyPercentEnd) - Number(seventyPercentStart)
+
+const statics = {
+    origin: origin
+}
+
+let vestings: DynamicVestingInfo[] = [data.rewards[0], data.rewards[1], data.rewards[2]].flatMap(reward => {
+        
+    const thirtyPercent = new BigNumber(reward.totalBsxReward).multipliedBy(new BigNumber(0.3)).toFixed(0)
+    const seventyPercent = new BigNumber(reward.totalBsxReward).multipliedBy(new BigNumber(0.7)).toFixed(0)
+    const perBlockBsxFor70percentVest = new BigNumber(seventyPercent).dividedBy(new BigNumber(leaseDuration)).toFixed(0)
+
+    return  [
+        {
+            destination: reward.address,
+            schedule: {
+                amtToBeVestedOverTime: thirtyPercent,
+                start: thirtyPercentStart,
+                period: '1',
+                per_period: thirtyPercent,
+                period_count: '1'
+            }
+            
+        },
+        {
+            destination: reward.address,
+            schedule: {
+                amtToBeVestedOverTime: seventyPercent,
+                start: seventyPercentStart,
+                period: '1',
+                per_period: perBlockBsxFor70percentVest,
+                period_count: leaseDuration.toString()
+            }
+        }
```

Comment:
here is where the 30% immediate-vesting-schedule,  and the 70% vesting-over-time schedule,  are created,  for each address,  according to what the [vesting pallet](https://github.com/open-web3-stack/open-runtime-module-library/blob/master/vesting/src/lib.rs) needs.  I'm new at reading rust code.

Reply:
I also believe the periods for the remaining 70% should last a week, not just a block. This would mean that you can only claim once a week cc @jak-pan to confirm

Reply:
latest commit I've implemented weekly periods

Reply:
Not sure. It will make bumps in price I think there is no reason not to go with daily or block based

Reply:
per block like we talked about in discord implemented with comments